angrmanagement/logic/jobmanager.py

```diff
@@ -71,28 +71,51 @@ def run(self) -> None:
             if any(job.blocking for job in self.job_manager.jobs) and GlobalInfo.main_window.isVisible():
                 gui_thread_schedule(GlobalInfo.main_window._progress_dialog.show, args=())
 
-            try:
-                self.current_job = job
-                ctx = JobContext(self.job_manager, job)
-                ctx.set_progress(0)
-
-                log.info('Job "%s" started', job.name)
-                job.start_at = time.time()
-                result = job.run(ctx, self.job_manager.workspace.main_instance)
-                now = time.time()
-                duration = now - job.start_at
-                log.info('Job "%s" completed after %.2f seconds', job.name, duration)
-
-                self.current_job = None
-            except (Exception, KeyboardInterrupt) as e:  # pylint: disable=broad-except
-                sys.last_traceback = e.__traceback__
-                self.current_job = None
-                log.exception('Exception while running job "%s":', job.name)
-                if self.job_manager.job_worker_exception_callback is not None:
-                    self.job_manager.job_worker_exception_callback(job, e)
+            # If job has cancelled attribute and it is True for cancelled, then skip it
+            if hasattr(job, "cancelled") and job.cancelled:
+                pass
             else:
-                self.job_manager.jobs.remove(job)
-                gui_thread_schedule_async(job.finish, args=(self.job_manager.workspace.main_instance, result))
+                try:
```

Comment:
Looks like most of the code between these two if/else branches is duplicated. Restructure it to avoid duplicating anything.